### PR TITLE
Initial draft of conda-libmamba-solver rollout blogpost.

### PR DESCRIPTION
**Pull request by jezdez**
### Description

A blog post to clarify out intentions for conda-libmamba-solver this year

### CODE REVIEW

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+description: |
+    Learn about the upcoming rollout plan for the conda-libmamba-solver, bringing enhanced performance and user experience to conda.
+authors: [jezdez]
+tags: [conda, mamba, solver, libmamba, dependency solver, performance, user experience, conda-forge, pycosat, plugin package, Anaconda Distribution, miniconda, CEP 8, CEP 9]
+
+---
+## Background
+
+
+Over a year ago the conda team at Anaconda, together with conda developers at Quansight and QuantStack, [announced the release](https://www.anaconda.com/blog/a-faster-conda-for-a-growing-community) of a new solver for conda based on the solver that the [mamba project](https://mamba.readthedocs.io/) had developed on top of the venerable libsolv project.
```
Comment (kathatherine) [inline blog/2023-06-10-conda-libmamba-solver-rollout.mdx:13]:
```suggestion
Over a year ago, the conda team at Anaconda, together with conda developers at Quansight and QuantStack, [announced the release](https://www.anaconda.com/blog/a-faster-conda-for-a-growing-community) of a new solver for conda based on the solver that the [mamba project](https://mamba.readthedocs.io/) had developed on top of the venerable libsolv project.
```

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+The dependency solver is one of the main pillars that make up the conda package and environment manager (next to network & IO stack and the command line interface integration for example) and has been the source of many performance and user experience issues over the years, given the growth of the conda ecosystem. Users of conda-forge and other larger conda channels in particular struggled with the current solver implementation based on pycosat.
```
Comment (kathatherine) [inline blog/2023-06-10-conda-libmamba-solver-rollout.mdx:15]:
```suggestion
The dependency solver is one of the main pillars that makes up the conda package and environment manager (next to the network & IO stack and the command line interface integration, for example) and has been the source of many performance and user experience issues over the years, given the growth of the conda ecosystem. Users of conda-forge and other larger conda channels in particular struggled with the current solver implementation based on pycosat.
```

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+[Fast-forward to December 2022](https://www.anaconda.com/blog/conda-is-fast-now), we’ve encouraged the use of the libmamba solver via the separate plugin package “conda-libmamba-solver" as a way to trial the use of the solver and get more feedback from real-world users. In case you haven't had the chance to provide your feedback, please head over to the [conda-libmamba-solver repository on GitHub](https://github.com/conda/conda-libmamba-solver).
```
Comment (kathatherine) [inline blog/2023-06-10-conda-libmamba-solver-rollout.mdx:17]:
```suggestion
[Fast-forward to December 2022](https://www.anaconda.com/blog/conda-is-fast-now); we’ve encouraged the use of the libmamba solver via the separate plugin package “conda-libmamba-solver" as a way to trial the use of the solver and get more feedback from real-world users. In case you haven't had the chance to provide your feedback, please head over to the [conda-libmamba-solver repository on GitHub](https://github.com/conda/conda-libmamba-solver).
```

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+## Roll-out Plan
+
+
+We're happy to announce the roll-out plan for the next phase for the solver, which we’re starting to implement to get the conda-libmamba-solver in the hands of more people.
+
+The June 2023 editions of the [Anaconda Distribution](https://www.anaconda.com/download/) and [miniconda](https://docs.conda.io/en/latest/miniconda.html) installers will include the [conda-libmamba-solver](https://conda.github.io/conda-libmamba-solver/) package by default for new installations.
+
+This will make switching to the solver much easier for end users as it will require only setting `CONDA_SOVLER=libmamba` (for one-time runs) or `conda config --set solver=libmamba` (for persistently switching via the `.condarc` file) and not require additional installation steps. More details on that in the [conda-libmamba-solver getting started guide](https://conda.github.io/conda-libmamba-solver/getting-started/).
```
Comment (kathatherine) [inline blog/2023-06-10-conda-libmamba-solver-rollout.mdx:26]:
```suggestion
This will make switching to the solver much easier for end users, as it will require only setting `CONDA_SOVLER=libmamba` (for one-time runs) or `conda config --set solver=libmamba` (for persistently switching via the `.condarc` file) and not require additional installation steps. More details on that in the [conda-libmamba-solver getting started guide](https://conda.github.io/conda-libmamba-solver/getting-started/).
```

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+We expect many more users to try it out that way to get more feedback where we can improve it further. If you’d like to send us your feedback now, please don’t hesitate to open ticket in the [conda](https://github.com/conda/conda/issues/new/choose) and [conda-libmamba-solver](https://github.com/conda/conda-libmamba-solver/issues/new/choose) repositories on GitHub accordingly.
```
Comment (kathatherine) [inline blog/2023-06-10-conda-libmamba-solver-rollout.mdx:28]:
```suggestion
We expect many more users to try it out that way to get more feedback where we can improve it further. If you’d like to send us your feedback now, please don’t hesitate to open a ticket in the [conda](https://github.com/conda/conda/issues/new/choose) and [conda-libmamba-solver](https://github.com/conda/conda-libmamba-solver/issues/new/choose) repositories on GitHub accordingly.
```

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+## Changing the default solver
+
+
+Afterwards, in the September 2023 release window for conda (`23.9.0` per [CEP 8](https://github.com/conda-incubator/ceps/blob/main/cep-8.md)), we'll also:
+
+
+- switch the default solver configuration to `libmamba`, no need to `--solver=libmamba` anymore
```
Comment (kathatherine) [inline blog/2023-06-10-conda-libmamba-solver-rollout.mdx:36]:
```suggestion
- switch the default solver configuration to `libmamba`(no need to `--solver=libmamba` anymore)
```

**Review by kathatherine — CHANGES_REQUESTED** [review]
A few grammar and punctuation fixes, but reads great to me!

**Comment by jezdez** [comment]
Thank you @kathatherine!

**Comment by jezdez** [comment]
We want to wait for Wolf and Jaime to take a look at it

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+The dependency solver is one of the main pillars that makes up the conda package and environment manager (next to the network & IO stack and the command line interface integration, for example) and has been the source of many performance and user experience issues over the years, given the growth of the conda ecosystem. Users of conda-forge and other larger conda channels in particular struggled with the current solver implementation based on pycosat.
+
+[Fast-forward to December 2022](https://www.anaconda.com/blog/conda-is-fast-now); we’ve encouraged the use of the libmamba solver via the separate plugin package “conda-libmamba-solver" as a way to trial the use of the solver and get more feedback from real-world users. In case you haven't had the chance to provide your feedback, please head over to the [conda-libmamba-solver repository on GitHub](https://github.com/conda/conda-libmamba-solver).
+
```
Comment (tnabtaf) [inline blog/2023-06-10-conda-libmamba-solver-rollout.mdx:18]: Suggest inserting a 

```
<!-- truncate -->
```

somewhere around this paragraph. Everything above that will appear on the main blog page, followed by a see more link that takes the reader to the full blog post

Comment (jezdez) [reply blog/2023-06-10-conda-libmamba-solver-rollout.mdx:18]: Good idea!

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+The dependency solver is one of the main pillars that makes up the conda package and environment manager (next to the network & IO stack and the command line interface integration, for example) and has been the source of many performance and user experience issues over the years, given the growth of the conda ecosystem. Users of conda-forge and other larger conda channels in particular struggled with the current solver implementation based on pycosat.
+
+[Fast-forward to December 2022](https://www.anaconda.com/blog/conda-is-fast-now); we’ve encouraged the use of the libmamba solver via the separate plugin package “conda-libmamba-solver" as a way to trial the use of the solver and get more feedback from real-world users. In case you haven't had the chance to provide your feedback, please head over to the [conda-libmamba-solver repository on GitHub](https://github.com/conda/conda-libmamba-solver).
```
Comment (jaimergp) [inline blog/2023-06-10-conda-libmamba-solver-rollout.mdx:14]: Alternative wording, a bit more in line with the suggestion above. Only a suggestion, feel free to ignore.

```suggestion
[In December 2022](https://www.anaconda.com/blog/conda-is-fast-now), we announced a new plugin package, `conda-libmamba-solver`, which integrates the `mamba` solver in `conda` itself. With this early release we aimed to obtain feedback from actual usage in real-world scenarios. If you have tried it and want to say something, we are still happy to hear from you in the [conda-libmamba-solver repository on GitHub](https://github.com/conda/conda-libmamba-solver).
```

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+## Roll-out Plan
+
+
+We're happy to announce the roll-out plan for the next phase for the solver, to get the conda-libmamba-solver in the hands of more people.
+
+<!-- truncate -->
```
Comment (jaimergp) [inline blog/2023-06-10-conda-libmamba-solver-rollout.mdx:21]: Isn't this truncation a bit too late in the text? I thought we were only using it for "first paragraph" intro. This is how it looks: https://deploy-preview-143--conda-dot-org.netlify.app/blog, which is honestly not that bad, but just in case the decision wasn't deliberate, I thought I'd bring it up.

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+The June 2023 editions of the [Anaconda Distribution](https://www.anaconda.com/download/) and [miniconda](https://docs.conda.io/en/latest/miniconda.html) installers will include the [conda-libmamba-solver](https://conda.github.io/conda-libmamba-solver/) package by default for new installations.
```
Comment (jaimergp) [inline blog/2023-06-10-conda-libmamba-solver-rollout.mdx:23]:
```suggestion
The June 2023 editions of the [Anaconda Distribution](https://www.anaconda.com/download/) and [miniconda](https://docs.conda.io/en/latest/miniconda.html) installers will include the [conda-libmamba-solver](https://conda.github.io/conda-libmamba-solver/) package in the `base` environment.
```

Comment (jaimergp) [reply blog/2023-06-10-conda-libmamba-solver-rollout.mdx:23]: Just wanted to remove the "default" term there because it can be confusing (shipped by default vs enabled by default).

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+This will make switching to the solver much easier for end users, as it will require only setting `CONDA_SOVLER=libmamba` (for one-time runs) or `conda config --set solver=libmamba` (for persistently switching via the `.condarc` file) and not require additional installation steps. More details on that in the [conda-libmamba-solver getting started guide](https://conda.github.io/conda-libmamba-solver/getting-started/).
```
Comment (jaimergp) [inline blog/2023-06-10-conda-libmamba-solver-rollout.mdx:25]: I'd mention the CLI flag instead of the env var.

```suggestion
This will make switching to the solver much easier for end users, as it will require only adding `--solver=libmamba` (for one-time runs) or `conda config --set solver=libmamba` (for persistently switching via the `.condarc` file) and not require additional installation steps. More details on that in the [conda-libmamba-solver getting started guide](https://conda.github.io/conda-libmamba-solver/getting-started/).
```

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+## Changing the default solver
+
+
+Afterwards, in the September 2023 release window for conda (`23.9.0` per [CEP 8](https://github.com/conda-incubator/ceps/blob/main/cep-8.md)), we'll also:
+
+
+- switch the default solver configuration to `libmamba`(no need to `--solver=libmamba` anymore)
+- keep the `classic` solver code included in conda, to be able to "switch back”, if circumstances require it, via `--solver=classic` and
+- create a plan to retire the `classic` solver in the far-future.
```
Comment (jaimergp) [inline blog/2023-06-10-conda-libmamba-solver-rollout.mdx:37]:
```suggestion
- switch the default solver configuration to `libmamba`(no need to `--solver=libmamba` anymore),
- keep the `classic` solver code included in `conda`, to be able to "switch back”, if circumstances require it, via `--solver=classic`, and
- create a plan to retire the `classic` solver in the far-future.
```

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+
+The long-term plan for the `classic` solver will include a timeline for deprecation from the main conda code base based on [CEP 9](https://github.com/conda-incubator/ceps/blob/main/cep-9.md) and additional public discussions with users if it's worth keeping the solver maintained in a separate conda plugin package (for example).
```
Comment (jaimergp) [inline blog/2023-06-10-conda-libmamba-solver-rollout.mdx:39]:
```suggestion
The long-term plan for the `classic` solver will include a timeline for deprecation from the main `conda` code base based on [CEP 9](https://github.com/conda-incubator/ceps/blob/main/cep-9.md) and additional public discussions with users if it's worth keeping the solver maintained in a separate conda plugin package (for example).
```

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+
+We understand that there are users who have built their workflows around the classic solver, and we want to provide ample time and support for them to migrate to the new solver. In case you have any input on our plans, please let us know! We look forward to working closely with the rest of the conda community to ensure a smooth transition.
```
Comment (jaimergp) [inline blog/2023-06-10-conda-libmamba-solver-rollout.mdx:41]: Lovely <3

**Review by jaimergp — APPROVED** [review]
Only a few comments regarding flow and simplifying some sentences. Only suggestions, feel free to ignore or readdress. Nothing I said should impede merging.

That said, I wonder if there's a TL;DR worth mentioning at either the beginning or the end. It's all about these three bullet points, after all:

- From June on, conda-libmamba-solver will be part of the official installers. The default solver is still `classic`.
- From September on, the default solver will be `libmamba`. `classic` will be available via `--solver=classic`, if needed.
- At some point in the future, we'll discuss whether the `classic` solver is kept around, maintained as a separate plugin, and/or deprecated for good. Feedback welcome!

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx [hunk 15]
```diff
@@ -0,0 +1,41 @@
+---
+title: "Conda and the libmamba solver: Roll-out plan 2023"
+slug: "2023-06-10-conda-libmamba-solver-rollout"
+description: |
+    Learn about the upcoming rollout plan for the conda-libmamba-solver, bringing enhanced performance and user experience to conda.
+authors: [jezdez]
+tags: [conda, mamba, solver, libmamba, dependency solver, performance, user experience, conda-forge, pycosat, plugin package, Anaconda Distribution, miniconda, CEP 8, CEP 9]
```
[the remark this answers is not in the document]
Comment (jezdez) [reply blog/2023-06-10-conda-libmamba-solver-rollout.mdx:7]: Sounds good, I was optimizing for too many keywords assuming they'd be useful for SEO reasons..

[the remark this answers is not in the document]
Comment (jezdez) [reply blog/2023-06-10-conda-libmamba-solver-rollout.mdx:7]:
```suggestion
tags: [conda-libmamba-solver, solvers, performance]
```

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx (hunk 9 above)
Comment (jezdez) [reply blog/2023-06-10-conda-libmamba-solver-rollout.mdx:21]: It feels like we're burying the lede if we're truncating earlier.

Comment (jezdez) [reply blog/2023-06-10-conda-libmamba-solver-rollout.mdx]: Okay, I've added the TLDR section and moved the truncating up

**Comment by jezdez** [comment]
> Only a few comments regarding flow and simplifying some sentences. Only suggestions, feel free to ignore or readdress. Nothing I said should impede merging.
> 
> That said, I wonder if there's a TL;DR worth mentioning at either the beginning or the end. It's all about these three bullet points, after all:
> 
> * From June on, conda-libmamba-solver will be part of the official installers. The default solver is still `classic`.
> * From September on, the default solver will be `libmamba`. `classic` will be available via `--solver=classic`, if needed.
> * At some point in the future, we'll discuss whether the `classic` solver is kept around, maintained as a separate plugin, and/or deprecated for good. Feedback welcome!

Thanks, I've just added a TLDR section, that solved the truncating issue as well.

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+description: |
+    Learn about the upcoming rollout plan for the conda-libmamba-solver, bringing enhanced performance and user experience to conda.
+authors: [jezdez, jaimergp]
+tags: [conda-libmamba-solver, solvers, performance]
+
+---
+
+Over a year ago, the conda team at Anaconda, together with conda developers at Quansight and QuantStack, [announced the release](https://www.anaconda.com/blog/a-faster-conda-for-a-growing-community) of a new solver for conda based on the libraries that the [mamba project](https://mamba.readthedocs.io/) had developed on top of the venerable libsolv project.
+
+## TLDR;
```
Comment (jaimergp) [inline blog/2023-06-10-conda-libmamba-solver-rollout.mdx:13]:
```suggestion
## TL;DR
```

The official way to write it seems to be with the semicolon in the middle. https://en.wikipedia.org/wiki/TL;DR

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+
+We expect many more users to try it out that way to get more feedback where we can improve it further. If you’d like to send us your feedback now, please don’t hesitate to open a ticket in the [conda](https://github.com/conda/conda/issues/new/choose) and [conda-libmamba-solver](https://github.com/conda/conda-libmamba-solver/issues/new/choose) repositories on GitHub accordingly.
+
+## Changing the default solver
+
+
+Afterwards, in the September 2023 release window for conda (`23.9.0` per [CEP 8](https://github.com/conda-incubator/ceps/blob/main/cep-8.md)), we'll also:
+
+
+- switch the default solver configuration to `libmamba`(no need to `--solver=libmamba` anymore),
```
Comment (jaimergp) [inline blog/2023-06-10-conda-libmamba-solver-rollout.mdx:46]:
```suggestion
- switch the default solver configuration to `libmamba` (no need to use the `--solver=libmamba` CLI flag anymore),
```

**Review by jaimergp — APPROVED** [review]
Two typos but other than that, good to go from my side!

File: blog/2023-06-10-conda-libmamba-solver-rollout.mdx (hunk 16 above)
Comment (jezdez) [reply blog/2023-06-10-conda-libmamba-solver-rollout.mdx:13]: OK!
